features/slack/assets/js/slack.js

```diff
@@ -0,0 +1,165 @@
+$('#slackchannels').on('click', 'span.close', function () {
+    $(this).fadeOut(100);
+    var row = $(this).parents('tr');
+    var newRow = "<tr><td colspan=\"2\"><div id=\"slack_placeholder\"/></tr></tr>";
+
+    newRow = $(newRow).insertAfter(row);
+    var placeholder = newRow.find("#slack_placeholder");
+
+    confirm_delete("Are you sure you want to delete this Slack channel?", placeholder, this, function() {
+        var self = $(this);
+        var id = $(this).attr("id").split("-")[1];
+        delete_slack_channel_for_event(get_current_event_id(), id, function(data) {
+            ($(self).parents('.channel-row')).remove();
+        });
+    }, function() {
+        $(this).fadeIn(100);
+        newRow.remove();
+    });
+
+});
+
+function get_channel_data(url, params, channels, success_callback, error_callback) {
+    $.getJSON(url, params, function(data) {
+        if (data.length === 0) {
+            success_callback(channels);
+        } else {
+            channels.push.apply(channels, data);
+            // API fetches 20 at a time
+            params.offset += 20;
```

Comment:
might want to define this as a constant somewhere, since this is subject to change

Reply:
I removed this function as it not required. It was dead code which I forgot to remove.